pkg/util/taints/taints.go

```diff
@@ -217,10 +217,10 @@ func addTaints(oldTaints []v1.Taint, newTaints *[]v1.Taint) bool {
 			}
 		}
 		if !existsInNew {
-			*newTaints = append(*newTaints, oldTaint)
+			addedTaints = append(addedTaints, oldTaint)
```

Comment:
I think this can be reorganized to move into where the set `existsInNew = true`, and remove that variable.

Reply:
Maybe not? I think it should traverse all the newTaints to know if the newTaints has the specific oldTaint. So if move into the for statement, I don't know how to write the logic？